tests/transform_dialect/matmul_fill_spec_pad_pack.mlir

```diff
@@ -1,7 +1,23 @@
-// RUN: iree-compile --iree-hal-target-backends=amd-aie --compile-to=executable-sources %S/../samples/pad_pack_pipeline_e2e.mlir | iree-opt --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-hal-translate-target-executable-variants{target=amd-aie})))" --iree-codegen-transform-dialect-library=%s
+// RUN: iree-opt --iree-transform-dialect-interpreter %s | FileCheck %s
 
 // This script shows an example lowering matmul through pad-pack pipeline for AIE device.
 
+func.func @matmul_i32() {
```

Comment:
Can the hal boilerplate be removed? I.e. can %arg0 be an tensor.empty() etc

Reply:
These hal ops are needed if we use `iree-opt --iree-transform-dialect-interpreter`.

Reply:
It's better to keep the hal ops. That is a better representation of how these transformations get lowered